File: python_leaoai/python_leaoai/services/ApplicationServices/Content/ProdutosService.py
from typing import TypeVar, Generic
from sqlalchemy.orm import Session
from services import ItemService
from repositories import Repository
from models import Produto  # Modelo SQLAlchemy do Produto
from services.generic_service import GenericService  # Serviço genérico base

T = TypeVar('T')

class ProdutosService(GenericService[Produto, Repository]):
    def __init__(self, db: Session, item_service: ItemService, repository: Repository):
        super().__init__(db, item_service, repository)
        self._db = db
        self._srvItem = item_service
        self._repository = repository

    # Aqui você pode adicionar métodos específicos de Produto, se necessário

File: Framework.Services/ApplicationServices/Content/ProdutosService.cs
using Data;
using Data.Models.Catalogo;
using Framework.Repositories;
using Framework.Services.ApplicationServices.Core;

namespace Framework.Services.ApplicationServices.Content
{
    public class ProdutosService : GenericService<Produto, IRepository<Produto>>
    {
        private readonly ApplicationDbContext _context;
        private readonly ItemService _srvItem;
        private readonly IRepository<Produto> _repository;

        public ProdutosService(ApplicationDbContext context, ItemService itemService, IRepository<Produto> repository) : base(context, itemService, repository)
        {
            _context = context;
            _srvItem = itemService;
            _repository = repository;
        }
    }
}
